### PR TITLE
Dialect: deprecate toplevelSeparator as unused

### PR DESCRIPTION
**Pull request by kitbellew**

### CODE REVIEW

**Review by tgodzik — APPROVED** [review]

File: scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala [hunk 1]
```diff
@@ -228,7 +227,7 @@ final class Dialect private (
       allowViewBounds = allowViewBounds,
       allowWithTypes = allowWithTypes,
       allowXmlLiterals = allowXmlLiterals,
-      toplevelSeparator = toplevelSeparator,
+      toplevelSeparator = null, // unused
```
Comment (dos65) [inline scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala:230]: Wouldn't it be enough just to mark it as deprecated?
There is a probability that somebody might get NPE. I understand that it's low but who knows.

Comment (kitbellew) [reply scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala:230]: i can use "" if you prefer.

Comment (dos65) [reply scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala:230]: maybe keep it as it is?
we don't do anything similar with other deprecated settings here. for example: `allowWithTypes`

Comment (kitbellew) [reply scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala:230]: ok, will do

**Pull request merged**